datacapturing/src/main/java/de/cyface/datacapturing/persistence/MeasurementPersistence.java

```diff
@@ -2,118 +2,100 @@
+import de.cyface.persistence.FileUtils;
 import de.cyface.persistence.MeasuringPointsContentProvider;
-import de.cyface.persistence.RotationPointTable;
+import de.cyface.persistence.NoSuchMeasurementException;
+import de.cyface.persistence.Persistence;
+import de.cyface.persistence.model.GeoLocation;
+import de.cyface.persistence.model.Measurement;
+import de.cyface.persistence.model.Vehicle;
+import de.cyface.persistence.serialization.*;
+import de.cyface.utils.Validate;
 
 /**
- * This class wraps the Cyface Android persistence API as required by the <code>DataCapturingListener</code> and its
- * delegate objects.
+ * This class extends the {@link Persistence} class by live-Capturing specific functionality.
  *
  * @author Klemens Muthmann
  * @author Armin Schnabel
- * @version 5.0.4
+ * @version 6.0.2
  * @since 2.0.0
  */
-public class MeasurementPersistence {
+public class MeasurementPersistence extends Persistence {
 
-    /**
-     * <code>ContentResolver</code> that provides access to the {@link MeasuringPointsContentProvider}.
-     */
-    private final ContentResolver resolver;
     /**
      * A threadPool to execute operations on their own background threads.
      */
     private ExecutorService threadPool;
-    /**
-     * The authority used to identify the Android content provider to persist data to or load it from.
-     */
-    private final String authority;
-
     /**
      * Caching the current measurement identifier, so we do not need to ask the database each time we require the
      * current measurement identifier. This is <code>null</code> if there is no running measurement or if we lost the
      * cache due to Android stopping the application hosting the data capturing service.
      */
     private Long currentMeasurementIdentifier;
+    /**
+     * The file to write the geolocation points to.
+     */
+    private GeoLocationsFile geoLocationsFile;
+    /**
+     * The file to write the acceleration points to.
+     */
+    private AccelerationsFile accelerationsFile;
+    /**
+     * The file to write the rotation points to.
+     */
+    private RotationsFile rotationsFile;
+    /**
+     * The file to write the direction points to.
+     */
+    private DirectionsFile directionsFile;
 
     /**
      * Creates a new completely initialized <code>MeasurementPersistence</code>.
      *
      * @param resolver <code>ContentResolver</code> that provides access to the {@link MeasuringPointsContentProvider}.
      * @param authority The authority used to identify the Android content provider to persist data to or load it from.
+     * @param context The {@link Context} required to locate the app's internal storage directory.
      */
-    public MeasurementPersistence(final @NonNull ContentResolver resolver, final @NonNull String authority) {
-        this.resolver = resolver;
+    public MeasurementPersistence(@NonNull final Context context, final @NonNull ContentResolver resolver,
```

Comment:
Ok, my suggestion: I think it's intuitive to:
* first use the @ flags (because the also are usually the first thing before a method declaration?)
* then final/static (just because ... alphabetical order - I don't know how else to decide this?)